src/dispatch/async_dispatcher.rs

```diff
@@ -62,17 +63,23 @@ where
     pub fn dispatch(&mut self) {
         let (snd, mut inner) = self.data.sender();
 
-        self.thread_pool.spawn(move || {
-            {
-                let world: &World = inner.world.borrow();
-
-                for stage in &mut inner.stages {
-                    stage.execute(world);
+        self.thread_pool
+            .read()
+            .unwrap()
+            .0
+            .as_ref()
+            .unwrap()
+            .spawn(move || {
+                {
+                    let world: &World = inner.world.borrow();
+
+                    for stage in &mut inner.stages {
+                        stage.execute(world);
+                    }
                 }
```

Comment:
It could be avoided, but this mean call the function `Self::create_thread_pool()` each time a dispatcher builder is created.

If there is nothing against it I can take rid of that Option.

Reply:
I think I am missing something here. You cannot just remove the scope?

Reply:
oh Sorry I miss understood it. Removed